86c246m16/feature/deploy-ec2-hosted-models

ori-small opened the pull request:
added hugging face download request and tracked download, and added cuda and python version to ModelVersion.

AmitMizrachi618 committed:
First commit for huggingface to storage new objects.
First commit for huggingface to storage new objects.

(#86c6e25nn)

AmitMizrachi618 committed:
Adding cuba and python version to Model, editing norman hug requests.
Adding cuba and python version to Model, editing norman hug requests.

(#86c6e25nn)

AmitMizrachi618 committed:
Small changes to huggingface objects, making them operational for a test run.
Small changes to huggingface objects, making them operational for a test run.

(#86c6e25nn)

AmitMizrachi618 committed:
Bugfix to HuggingFaceDownloadRequest which has =str rather than :str.
Bugfix to HuggingFaceDownloadRequest which has =str rather than :str.

(#86c6e25nn)

AmitMizrachi618 committed:
Adding asset_type to ModelAsset class, to differenciate between requirements and inference py files.
Adding asset_type to ModelAsset class, to differenciate between requirements and inference py files.

(#86c6e25nn)

AmitMizrachi618 committed:
Adding HuggingFace and Git to receive format and input type, bug fix where we've had coda written instead of cuda.
Adding HuggingFace and Git to receive format and input type, bug fix where we've had coda written instead of cuda.

(#86c6e25nn)

AmitMizrachi618 committed:
Adding properties to Huggingface download request and tracked download.
Adding properties to Huggingface download request and tracked download.

(#86c6e25nn)

Dolevg2411 committed:
Adding cuda and python version and asset type.

(#86c6e265r)

Dolevg2411 committed:
Rename asset type to asset name.

(#86c6e265r)

Dolevg2411 committed:
Rename asset type to asset name.

(#86c6e265r)

Dolevg2411 committed:
Rename asset type to asset name.

(#86c6e265r)

ori-small committed:
Added addition status flag names according to DB changes.
Added addition status flag names according to DB changes.

(#86c246m16)

ori-small committed:
Rebased and merged changes after Dolev's local changes.
Rebased and merged changes after Dolev's local changes.

(#86c246m16)

ori-small committed:
Added efs permanent flag for file transfer to efs permanent location.
Added efs permanent flag for file transfer to efs permanent location.

(#86c246m16)

ori-small committed:
Fices all hugging face files to support the new flag.
Fices all hugging face files to support the new flag.

(#86c246m16)

ori-small committed:
Reverted changes in flags as they are not needed anymore.
Reverted changes in flags as they are not needed anymore.

(#86c246m16)

ori-small committed:
Added the permanent flag for efs model files transfer.
Added the permanent flag for efs model files transfer.

(#86c246m16)

ori-small committed:
Reverted flag changes according to current sns -> sqs subscriptions and communication.
Reverted flag changes according to current sns -> sqs subscriptions and communication.

(#86c246m16)

ori-small committed:
Changed paths according to the new EFS structure updates.
Changed paths according to the new EFS structure updates.

(#86c246m16)

Dolevg2411 committed:
Added readme to norman objects

ori-small committed:
CR fixes: removed permanent flags as it is not used by action track.
CR fixes: removed permanent flags as it is not used by action track.

(#86c246m16)

ori-small committed:
Added version_id to the EFs new hirarchy according to new model schema.
Added version_id to the EFs new hirarchy according to new model schema.

(#86c246m16)

ori-small committed:
Added version_id to huggingface_download_request.py for adapting to the new model hirarchy.
Added version_id to huggingface_download_request.py for adapting to the new model hirarchy.

(#86c246m16)

ori-small committed:
Removed duplicates of asset_id from huggingface_download_request.py.
Removed duplicates of asset_id from huggingface_download_request.py.

(#86c246m16)

ori-small committed:
Removed Model field to become ModelProjection for a single version handling.
Removed Model field to become ModelProjection for a single version handling.

(#86c246m16)

ori-small committed:
Added cuda and python version to model_version, and removed it from model.
Added cuda and python version to model_version, and removed it from model.

(#86c246m16)

ori-small committed:
Added the model id to hugging face tracked message.
Added the model id to hugging face tracked message.

(#86c246m16)

ori-small committed:
Changed asset message to get the asset i d in entity id field.
Changed asset message to get the asset i d in entity id field.

(#86c246m16)

AmitMizrachi618 committed:
Adding the licence file and updating README documentation to be short and concise.
Adding the licence file and updating README documentation to be short and concise.

(#86c71nd3q)

ori-small committed:
Changed flag name to contain asset_name, model_id not needed.
Changed flag name to contain asset_name, model_id not needed.

(#86c730dbp)

AmitMizrachi618 committed:
CR Wording fixes to objects readme documentation. small fixes.
CR Wording fixes to objects readme documentation. small fixes.

(#86c71nd3q)

ori-small committed:
Removed unnecessary changes + ordered the enum alphabetically.
Removed unnecessary changes + ordered the enum alphabetically.

(#86c730dbp)

ori-small committed:
CR fixes: changed huggingface download request to be a standalone class (no need for links and checksums), and changed naming according to cr.
CR fixes: changed huggingface download request to be a standalone class (no need for links and checksums), and changed naming according to cr.

(#86c730dbp)

AmitMizrachi618 committed:
Wording: Removing the word Overview from the headers, and the words core sdk from the bottom.
Wording: Removing the word Overview from the headers, and the words core sdk from the bottom.

(#86c71nd3q)

yonatankarimish committed:
Merge pull request #80 from norman-ml/86c71nd3q/feature/readme

Production-grade documentation for Norman SDK and dependency libraries.

(#86c71nd3q)

ori-small committed:
CR fixes: created shared/model_assets folder for better separation.
CR fixes: created shared/model_assets folder for better separation.

(#86c730dbp)

ori-small committed:
CR fixes: Added deployment -> image config to objects as it is used in several places.
CR fixes: Added deployment -> image config to objects as it is used in several places.

(#86c730dbp)

yonatankarimish committed:
Invocations, Inputs, Outputs now in an ordered class hierarchy.

Invocations, Inputs, Outputs can now resolve their own path on disk and in s3.

PathContext context variable static class created to serve the new path aware getters.

(#86c246m16)

ori-small committed:
CR fixes: added os_version to model version entity.
CR fixes: added os_version to model version entity.

(#86c246m16)

ori-small committed:
CR fixes: changed entity type to take directly from the enum as it is not a filed (not using self).
CR fixes: changed entity type to take directly from the enum as it is not a filed (not using self).

(#86c246m16)

yonatankarimish commented on norman_objects/shared/models/model_version.py:
```diff
@@ -27,6 +27,10 @@ class ModelVersion(ModelVersionPreview):
 
+    cuda_version: str
+    python_version: str
+    os_version: str
```
Naming: we need to rename to `ubuntu_version`

yonatankarimish committed:
Wrote convenience methods around file and bucket path getter methods for invocation signature and model asset classes.

(#86c246m16)

yonatankarimish committed:
Added staging path polymorphic method to file pull tracked downloads.

(#86c246m16)

yonatankarimish committed:
Added build path to model assets.

(#86c246m16)

yonatankarimish committed:
Added staging paths to queue messages.

(#86c246m16)

yonatankarimish committed:
Added staging paths to queue messages.

(#86c246m16)

ori-small committed:
CR fixes: changed os_version to be ubuntu_version.
CR fixes: changed os_version to be ubuntu_version.

(#86c246m16)

yonatankarimish committed:
Fixed a bug in which the build path partition name was not statically defined as the build partition.

(#86c246m16)

yonatankarimish committed:
Invocation signatures now specify signature type explicitly to match model signatures.

(#86c246m16)

yonatankarimish committed:
Modified model type huggingface constant name to a broader literal.

(#86c246m16)

ori-small committed:
CR fixes: removed duplicate of bucket name in storage path.
CR fixes: removed duplicate of bucket name in storage path.

(#86c246m16)

ori-small committed:
CR fixes: also deleted data bucket name in invocation signature.
CR fixes: also deleted data bucket name in invocation signature.

(#86c246m16)

ori-small committed:
CR fixes: reverted changes in asset bucket and data bucket name.
CR fixes: reverted changes in asset bucket and data bucket name.

(#86c246m16)

yonatankarimish committed:
Renamed huggingface model name to model name to avoid confusion and other devs assuming these are separate.

(#86c246m16)

yonatankarimish committed:
Renamed huggingface tracked download file to tracked huggingface download to match the rest of the tracked downloads.

(#86c246m16)

yonatankarimish committed:
...And also the class name

(#86c246m16)

ori-small committed:
CR fixes: removed buckets getters and setters as they are not needed.
CR fixes: removed buckets getters and setters as they are not needed.

(#86c246m16)

yonatankarimish committed:
Removed incorrect huggingface receive format literal from the enum.

(#86c246m16)

yonatankarimish committed:
Consolidated asset status flags into a single family.

(#86c246m16)

yonatankarimish reviewed (approved):
Approve, pending tests
